container-api/src/main/java/org/jboss/forge/furnace/repositories/AddonStorageRepository.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2012 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Eclipse Public License version 1.0, available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.jboss.forge.furnace.repositories;
+
+import org.jboss.forge.furnace.addons.Addon;
+import org.jboss.forge.furnace.addons.AddonId;
+
+import java.io.File;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Used to perform {@link Addon} installation/registration operations.
+ * 
+ * @author <a href="mailto:lincolnbaxter@gmail.com">Lincoln Baxter, III</a>
+ * @author <a href="mailto:koen.aers@gmail.com">Koen Aers</a>
+ * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
+ */
+public interface AddonStorageRepository
```

Comment:
I think we have a naming issue here :) 

Can you describe what Repository is? I'm trying to follow semantic of "Repository" as it stated in projects like Spring Data, but looks like in Furnace it's more like a business object, right?